pkg/backends/newrelic/newrelic.go

```diff
@@ -314,7 +315,10 @@ func (n *Client) post(ctx context.Context, buffer *bytes.Buffer, data interface{
 			return fmt.Errorf("[%s] %v", BackendName, err)
 		}
 
-		log.Warnf("[%s] failed to send, sleeping for %s: %v", BackendName, next, err)
+		n.logger.WithFields(logrus.Fields{
```

Comment:
Missing backend name?

Reply:
Backend name is now injected higher up in the system, in `InitBackend`.  The `logger` passed already has a name.

When everything is enabled it looks something like this wondrous monstrosity:
```
INFO[0000] Starting server
INFO[0000] No cloud provider specified
INFO[0000] created transport                             dialer-keep-alive=30s dialer-timeout=5s enable-http2=false idle-connection-timeout=1m0s max-idle-connections=50 name=default network=tcp tls-handshake-timeout=3s
INFO[0000] created client                                client-timeout=10s name=default type=http
INFO[0000] created backend                               backend=datadog compress-payload=true max-request-elapsed-time=15s max-requests=24 metrics-per-batch=1000
INFO[0000] Initialised backend                           backend=datadog
INFO[0000] created backend                               address="localhost:2003" backend=graphite counter-namespace=stats.counters dial-timeout=5s gauges-namespace=stats.gauges global-suffix= mode=tags sets-namespace=stats.sets timer-namespace=stats.timers write-timeout=30s
INFO[0000] Initialised backend                           backend=graphite
INFO[0000] Initialised backend                           backend=null
INFO[0000] created backend                               address="127.0.0.1:1234" backend=statsdaemon dial-timeout=5s write-timeout=30s
INFO[0000] Initialised backend                           backend=statsdaemon
INFO[0000] Initialised backend                           backend=stdout
INFO[0000] Initialised backend                           backend=cloudwatch
INFO[0000] internal metrics OFF, to enable set 'statser-type' to 'logging' or 'internal'  backend=newrelic
INFO[0000] flushInterval default (10s)                   backend=newrelic
INFO[0000] created backend                               backend=newrelic flush-interval=10s max-request-elapsed-time=15s max-requests=24 metrics-per-batch=1000
INFO[0000] Initialised backend                           backend=newrelic
INFO[0000] created transport                             dialer-keep-alive=2s dialer-timeout=30s enable-http2=false idle-connection-timeout=4s max-idle-connections=5 name=influxdb network=tcp4 tls-handshake-timeout=6s
INFO[0000] created client                                client-timeout=10s name=influxdb type=http
INFO[0000] created backend                               api-endpoint="http://influxdb-1.8.0.local/" api-version=2 backend=influxdb bucket=derp compress-payload=false credentials="(set)" max-request-elapsed-time=15s max-requests=120 metrics-per-batch=5000 org=org transport=influxdb user-agent=gostatsd
INFO[0000] Initialised backend                           backend=influxdb
```

Reply:
Neat